packages/esm-appointments-app/src/workload/monthly-view-workload/monthly-view.component.tsx

```diff
@@ -0,0 +1,54 @@
+import dayjs from 'dayjs';
+import React, { useContext } from 'react';
+import { useTranslation } from 'react-i18next';
+import DaysOfWeekCard from '../../calendar/monthly/days-of-week.component';
+import { monthDays } from '../../helpers';
+import SelectedDateContext from '../../hooks/selectedDateContext';
+import styles from './monthly-workload.scss';
+import MonthlyWorkloadCard from './monthlyWorkCard';
+
+interface MonthlyCalendarViewProps {
+  calendarWorkload: Array<{ count: number; date: string }>;
+  dateToDisplay?: string;
+}
+const daysInWeek = ['SUN', 'MON', 'TUE', 'WED', 'THUR', 'FRI', 'SAT'];
+const MonthlyCalendarView: React.FC<MonthlyCalendarViewProps> = ({ calendarWorkload, dateToDisplay = '' }) => {
+  const { selectedDate, setSelectedDate } = useContext(SelectedDateContext);
+
+  const monthViewDate = dateToDisplay === '' ? selectedDate : dateToDisplay;
+  const handleDivClick = (date) => {
+    setSelectedDate(date);
```

Comment:
So I think what we are saying here is that clicking on a day from within the day or month few will update the entire date context, while I'd think it would only want to change the appointment date?  That being said, I'm happy if we want to merge this in and see how it works.

Reply:
Modified clicking on a day from within the month to change the appointment date
![appointmentCalendarScheduler](https://github.com/openmrs/openmrs-esm-patient-management/assets/8075969/d8d2ab66-c2d0-4a47-b512-ae1479abd0fe)

Reply:
@gabriel090 can you try what happens if you click on a date in the calendar when using this form from the "Appointments" app (not the appointments view in the patient chart?)